### ISSUE TITLE
PDF export images broken, HTML works fine?

### ISSUE THREAD

**Issue by distressmess** [BookStackApp/BookStack#1568]
I've cross-posted this to the solidnerd docker version of Bookstack, but just in case this is an issue sourced with dompdf or Bookstack itself:

I'm having an issue with exporting PDFs in Bookstack. I'm not sure if replacing dompdf with wkhtmltopdf might help, but I can't seem to figure out how to modify the .env as described on the Bookstack site, https://www.bookstackapp.com/docs/admin/pdf-rendering/ , while using the whole solidnerd Bookstack image, so I'm unable to test if that's the issue at the moment. 

I'm running Windows 10 and Docker for Windows, as well as the solidnerd Bookstack image. 

PDFs render without images, instead only giving the link to the image. This error doesn't carry over to the html export.

See below for an image example:

![ruler](https://user-images.githubusercontent.com/53479820/62295338-cce25a80-b43a-11e9-9b9c-0f84567b0187.jpg)

Thanks in advance!

**Comment by distressmess** [comment]
As an update, I've tried using different image types (png, jpg) with no change. I've also tried to bring in wkhtmltopdf in a few different ways with no success, though that is probably down to my lack of experience in the field...

**Comment by ssddanbrown** [comment]
Hi @distressmess,
Sorry to hear your are experiencing issues.

I don't thinking switching WKHTML would solve this particular issue since the default DOMPDF should render images fine.

I'm assuming that BookStack is having troubling finding the images when it attempts to embed them when generating the HTML that's fed into the PDF generation library. 

Have you changed the storage mechanism at all or are you using the default `local` storage option for images? 
Have you set the `APP_URL` parameter in the `.env` file? Setting this can help in locating local image files.

**Comment by distressmess** [comment]
Hi Dan! The local storage mechanism is the same as the default. I believe I set app_url to http://localhost, but I'll double check with a fresh install and see if that does anything.

**Comment by Dayflare** [comment]
i have a similar issue with pictures in the exported pdf document. The pictures are displayed but they're put under the text so the text is written over the pictures.

the first screenshot shows the page in bookstack, the second one in pdf.
![msedge_2019-08-08_14-07-44](https://user-images.githubusercontent.com/42730126/62702093-2b34ad80-b9e6-11e9-9aaa-634707697f0f.png)
![AcroRd32_2019-08-08_14-08-15](https://user-images.githubusercontent.com/42730126/62702104-2f60cb00-b9e6-11e9-83a5-c0f4f220859b.png)

maybe this has something to do with @distressmess issue.

**Comment by distressmess** [comment]
Ahhh, I solved it! Partially inspired by issue #799 (https://github.com/BookStackApp/BookStack/issues/799)

For future reference of anyone with the same issue, I followed the steps on the Solidnerd guide, but since I was a little confused by how to actually call the containers, I typed them in full lines in PowerShell while doing a fresh install of the Bookstack setup:

1. docker pull solidnerd/bookstack
2. docker network create bookstack_nw
3. docker run -d --net bookstack_nw -e MYSQL_ROOT_PASSWORD=secret -e MYSQL_DATABASE=bookstack -e MYSQL_USER=bookstack -e MYSQL_PASSWORD=secret --name="bookstack_db" mysql:5.7.21
4. docker run -d --net bookstack_nw -e DB_HOST=bookstack_db:3306 -e DB_DATABASE=bookstack -e DB_USERNAME=bookstack -e DB_PASSWORD=secret -e APP_URL=http://127.0.0.1:8080 -p 8080:80 solidnerd/bookstack:latest
5. docker ps -a
6. docker start (whatever the name of your database is)
7. docker start (whatever the name of your bookstack instance is)

And then go to your app url to find a working copy of Bookstack with images functioning in the PDF exports! 

Hopefully these instructions are easy enough that anyone with the same issue later and similar noob-levels of sysadmin stuff can make it work. :D
